src/jimgw/core/single_event/likelihood.py

```
@@ -385,6 +389,134 @@ def _likelihood(self, params: dict[str, Float], data: dict) -> Float:
+
+        if "d_L" not in dist_prior.parameter_names:
+            raise ValueError(
+                "The provided prior does not contain a sub-prior for 'd_L'. "
+                "Ensure your prior has parameter_names=['d_L']."
+            )
+
+        if not hasattr(dist_prior, "xmin") or not hasattr(dist_prior, "xmax"):
+            raise ValueError(
+                "The d_L sub-prior must have xmin and xmax attributes. "
+                "Use a bounded prior such as PowerLawPrior or UniformPrior."
+            )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)